chisel/src/main.rs

```diff
@@ -345,6 +255,49 @@ fn chisel_execute(context: &ChiselContext) -> Result<bool, &'static str> {
     }
 }
 
+fn yaml_configure(yaml: &str) -> ChiselConfig {
+    if let Ok(rulesets) = serde_yaml::from_str::<Value>(yaml) {
+        let config = ChiselConfig::from_yaml(&rulesets);
+        if config.is_err() {
+            err_exit(&config.unwrap_err());
+        } else {
+            config.unwrap()
+        }
+    } else {
+        err_exit("Failed to parse YAML configuration")
+    }
+}
+
+impl From<&(String, config::Ruleset)> for ChiselContext {
+    fn from(input: &(String, config::Ruleset)) -> ChiselContext {
```

Comment:
Do we want to take a reference or can we simply take it by value and consume it?

Reply:
That needs a change in the caller, this was the simplest approach.

Reply:
It is actually worse trying to do that:
```
320 |             let ctx: ChiselContext = (*ctx).into();
    |                                      ^^^^^^ move occurs because `*ctx` has type `(std::string::String, config::Ruleset)`, which does not implement the `Copy` trait
```